datafusion/core/src/physical_plan/windows/mod.rs

```diff
@@ -187,6 +188,30 @@ fn create_built_in_window_expr(
     })
 }
 
+pub(crate) fn calc_requirements(
+    partition_by_exprs: &[Arc<dyn PhysicalExpr>],
+    orderby_sort_exprs: &[PhysicalSortExpr],
+) -> Option<Vec<PhysicalSortRequirement>> {
+    let mut sort_reqs = vec![];
+    for partition_by in partition_by_exprs {
+        sort_reqs.push(PhysicalSortRequirement {
+            expr: partition_by.clone(),
+            options: None,
+        });
+    }
+    for PhysicalSortExpr { expr, options } in orderby_sort_exprs {
+        let contains = sort_reqs.iter().any(|e| expr.eq(&e.expr));
+        if !contains {
+            sort_reqs.push(PhysicalSortRequirement {
+                expr: expr.clone(),
+                options: Some(*options),
+            });
+        }
+    }
+    // Convert empty result to None. Otherwise wrap result inside Some()
+    (!sort_reqs.is_empty()).then_some(sort_reqs)
+}
+
```

Comment:
If there is overlap between partition by keys and sort keys, I think we should respect the sort key's requirements.

Reply:
In the case of `PARTITION BY a, ORDER BY a`. We do not add requirement for column `a`. The reason is that all partition would consist of the same value of `a`. Hence `ORDER BY` doesn't really define a direction(all `a` values would be same.). Above operation can produce correct result when its input is ordered according `a ASC` or `a DESC`.

Reply:
Since `ORDER BY` is local to each partition, I think we are fine here. There is some related discussion here: https://stackoverflow.com/questions/50364818/using-the-same-column-in-partition-by-and-order-by-with-dense-rank

I am still taking a note to remind us that we may want to revisit this if we find out information indicating otherwise.

Reply:
Computing `rownum()` over  `PARTITION BY a, ORDER BY a DESC`  I think will result in arbitrary assignments of row numbers (as @ozankabak  and @mustafasrepo  say above, the value of `a` within each partition is the same)